src/Components/SearchBar/Searchbar.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import { useAppDispatch, useAppSelector } from 'features/reduxHooks';
+import ToggleAdvantageDisadvantage from 'Components/ToggleAdvantageDisadvantage/ToggleAdvantageDisadvantage';
+import { storeAdvantageList, storeDisadvantageList } from 'features/characterSlice';
+import Select from 'react-select';
+import { SelectOptionObj } from '../../../types';
+import AdvantagesArray from '../../Attribute Objects/Advantages';
+import DisadvantagesArray from '../../Attribute Objects/Disadvantages';
+
+import './Searchbar.styles.scss';
+import { SearchbarTitle } from './SearchbarTitle';
+
+const Searchbar = () => {
+  const dispatch = useAppDispatch();
+  const selectedAdvantages = useAppSelector((state) => state.character.advantages);
+  const selectedDisadvantages = useAppSelector((state) => state.character.disadvantages);
+
+  const isChoosingAdvantages = useAppSelector((state) => state.toggle.isChoosingAdvantages);
+
+  const advantageOptions: SelectOptionObj[] = AdvantagesArray.map((opt) => ({
+    label: opt.title,
+    value: opt.title,
+    category: opt.type
+  }));
+  const disadvantageOptions: SelectOptionObj[] = DisadvantagesArray.map((opt) => ({
+    label: opt.title,
+    value: opt.title,
+    category: opt.type
+  }));
+
+  let selectedOptions: SelectOptionObj[] = [];
+  const handleChange = async (e: SelectOptionObj[]) => {
+    selectedOptions = e;
+
+    const selectedAdvantages = e
```

Comment:
Loving your use of array methods and deriving state, looks like you've really got them down

Reply:
Thanks man.
The other day i looked at the last way I was this and wondered what the hell I was thinking.
I way over complicated it, much cleaner now.